src/types.rs

```diff
@@ -54,10 +62,30 @@ pub struct ExtremePoint<T>
     pub xmin: Point<T>,
 }
 
+/// A single Point in 2D space.
+///
+/// Points can be created using the `new(x, y)` constructor, or from a `Coordinate` or pair of points.
+///
+/// ```
```

Comment:
Examples sections [should have an `# Examples` heading](https://github.com/rust-lang/rfcs/blob/master/text/1574-more-api-documentation-conventions.md#examples-in-api-docs) before the code blocks, though it's not a big deal such that we need to hold up this PR